ToolChain/Assembler/Parser/Parser.py

```diff
@@ -446,20 +346,80 @@ def _evaluateAndExtractInfoFromLine(self, line, buildInstruction):
                     part = part[1:]
                 if not foundSource:
                     foundSource = True
+
                     # Verify this reference doesn't have additonal ':', hotfix for issue 18
                     if part.count(':') > 0:
                         raise ValueError("Syntax error, memory reference has too many {}".format(MEMORY_REFERENCE_INDICATORS))
-                    buildInstruction.sourceImmediate = MEMORY_REFERENCE_INDICATORS+ part.upper() + MEMORY_REFERENCE_INDICATORS
+                    buildInstruction.sourceImmediate = MEMORY_REFERENCE_INDICATORS + part.upper() + \
+                                                       MEMORY_REFERENCE_INDICATORS
+                    
                 else:
                     if foundDestination:
                         raise ValueError("Invalid operation format")
                     foundDestination = True
+
                     # Verify this label doesn't have additonal ':', hotfix for issue 18
                     if part.count(':') > 0:
```

Comment:
I know this is not new code but this comment applies. If we are updating the file I think it should be cleaned up
":" should be replaced with MEMORY_REFERENCE_INDICATORS